### PR TITLE
Tunnel Scoped TC MAP and Remarking

### PR DESCRIPTION
**Pull request by JaiOCP**
This PR add following behavior to tunnel encap and decap

Tunnel Encap:
1.      SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP for remarking the packet based on normalized TC as per port TC MAP binding
2.      SAI_TUNNEL_ATTR_ENCAP_QOS_TC_TO_QUEUE_MAP for queue selection based on normalized TC as per port TC MAP binding

Tunnel Decap:
1.      SAI_TUNNEL_ATTR_DECAP_QOS_DSCP_TO_TC_MAP new TC MAP can be bind'ed to the tunnel for decap. This MAP overrides port MAP if configured.
2.      SAI_TUNNEL_ATTR_DECAP_QOS_TC_TO_PRIORITY_GROUP_MAP for mapping the normalized TC to priority group mainly for PFC generation.

DSCP mode takes effect as configured for both cases.

### CODE REVIEW

**Review by kcudnik — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Comment by rlhui** [comment]
@ashutosh-agrawal  - would you be approving this? Thanks.

File: inc/saitunnel.h [hunk 1]
```diff
@@ -686,6 +686,56 @@ typedef enum _sai_tunnel_attr_t
...
+
+    /**
+     * @brief Enable DSCP -> TC MAP on tunnel at termination (Network-to-access) node. This map if configured overrides the port MAP
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_QOS_MAP
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
```
Comment (rck-innovium) [inline inc/saitunnel.h:722]: It is not clear which DSCP we are referring here? Is it from the payload or from the tunnel header?

Comment (JaiOCP) [reply inc/saitunnel.h:722]: DSCP is based on configured the configured DSCP mode for the tunnel. For the usecase discussed it will be uniform mode.

**Review by ashutosh-agrawal — APPROVED** [review]

File: inc/saitunnel.h [hunk 2]
```diff
@@ -1082,6 +1132,8 @@ typedef struct _sai_tunnel_api_t
     sai_remove_tunnel_map_entry_fn               remove_tunnel_map_entry;
     sai_set_tunnel_map_entry_attribute_fn        set_tunnel_map_entry_attribute;
     sai_get_tunnel_map_entry_attribute_fn        get_tunnel_map_entry_attribute;
+    sai_bulk_object_get_attribute_fn             get_tunnels_attribute;
```
[the remark this answers is not in the document]
Comment (kcudnik) [reply inc/saitunnel.h:1135]: added https://github.com/opencomputeproject/SAI/pull/1462

[the remark this answers is not in the document]
Comment (JaiOCP) [reply inc/saitunnel.h:1135]: Thanks

**Pull request merged**